manager/model.go

```diff
@@ -182,8 +182,8 @@ type SessionQuerier interface {
 	NewSession() Session
 	NewNoopSession() Session
 	GetSession(uuid.UUID) Session
-	WithSession(Session, func(Session) error) error
-	WithNewSession(f func(Session) error) error
+	WithSession(Session, func(Session) error, bool) error
```

Comment:
Don't break the interface. You can create a "close" property inside sessionManager implementation and pass it in the Ctor (newSessionManager).